rs/rosetta-api/icrc1/ledger/tests/tests.rs

```diff
@@ -1543,3 +1553,151 @@ mod verify_written_blocks {
+            ledger_id,
+            account(1).owner,
+            &ApproveArgs {
+                from_subaccount: None,
+                spender: account(2),
+                amount: Nat::from(big_amount),
+                expected_allowance: None,
+                expires_at: None,
+                fee: None,
+                memo: None,
+                created_at_time: None,
+            },
+        );
+        assert_eq!(approval_result, Ok(BlockIndex::from(1u64)));
+        balance_1 -= FEE;
+
+        // Try to upgrade the ledger from using a u64 wasm to a u256 wasm
+        let upgrade_args = Encode!(&LedgerArgument::Upgrade(None)).unwrap();
+        env.upgrade_canister(ledger_id, ledger_mainnet_u256_wasm(), upgrade_args)
+            .expect("Unable to upgrade the ledger canister");
+
+        // The balance and allowance should not change
+        assert_eq!(balance_1, balance_of(&env, ledger_id, account(1)));
+        let allowance_2 = get_allowance(&env, ledger_id, account(1), account(2));
+        assert_eq!(allowance_2.allowance, Nat::from(u64::MAX));
+
+        // Try to upgrade the ledger back to a u64 wasm
+        let upgrade_args = Encode!(&LedgerArgument::Upgrade(None)).unwrap();
+        env.upgrade_canister(ledger_id, ledger_mainnet_u64_wasm(), upgrade_args)
+            .expect("Unable to upgrade the ledger canister");
+
+        // The balance and allowance should not change
+        assert_eq!(balance_1, balance_of(&env, ledger_id, account(1)));
+        assert_eq!(allowance_2.allowance, Nat::from(u64::MAX));
+    }
+
+    #[test]
+    fn should_fail_to_upgrade_to_u256_ledger_wasm_if_balance_too_large() {
+        let env = StateMachine::new();
+        let ledger_id = env
+            .install_canister(ledger_mainnet_u256_wasm(), default_init_args(), None)
+            .unwrap();
+        transfer(&env, ledger_id, MINTER, account(1), u64::MAX - 100);
+        transfer(&env, ledger_id, MINTER, account(2), u64::MAX - 100);
+        let balance_1 = balance_of(&env, ledger_id, account(1));
+        let balance_2 = balance_of(&env, ledger_id, account(2));
+        assert_eq!(balance_1, balance_of(&env, ledger_id, account(1)));
+        assert_eq!(balance_2, balance_of(&env, ledger_id, account(2)));
+
+        let upgrade_args = Encode!(&LedgerArgument::Upgrade(None)).unwrap();
+        assert_matches!(
+            env.upgrade_canister(ledger_id, ledger_mainnet_u64_wasm(), upgrade_args),
+            Err(err)
+            if err.code() == CanisterCalledTrap
+              && err.description().contains("invalid type: enum, expected u64 or { e8s: u64 } struct")
+        );
+    }
+
+    #[test]
+    fn should_fail_to_upgrade_to_u256_ledger_wasm_if_allowance_too_large() {
+        let env = StateMachine::new();
+        let ledger_id = env
+            .install_canister(ledger_mainnet_u256_wasm(), default_init_args(), None)
+            .unwrap();
+        transfer(&env, ledger_id, MINTER, account(1), 1_000_000);
```

Comment:
nit: Maybe choose an amount having a relation with `FEE`, e.g.
```suggestion
        transfer(&env, ledger_id, MINTER, account(1), FEE);
```

Reply:
Makes sense, fixed!